pygments/lexers/dotnet.py

```diff
@@ -727,3 +727,142 @@ def analyse_text(text):
             result += 0.05
 
         return result
+
+class XppLexer(RegexLexer):
+
+    """
+    For X++ source code. this is based loosely on the CSharpLexer
+    Additional options accepted:
+
+    ``unicodelevel``
+       Determines which Unicode characters this lexer allows for identifiers.
+       The possible values are:
+
+       * ``none`` -- only the ASCII letters and numbers are allowed. This
+         is the fastest selection.
+       * ``basic`` -- all Unicode characters from the specification except
+         category ``Lo`` are allowed.
+       * ``full`` -- all Unicode characters as specified in the C# specs
+         are allowed.  Note that this means a considerable slowdown since the
+         ``Lo`` category has more than 40,000 characters in it!
+
+       The default value is ``basic``.
```

Comment:
Odd so your results for full was actually lower than none or basic? That would not be expected. Honestly I used the same set of characters as the C# lexer because in theory we support the same characters so I thought it would be necessary to have the option.

Reply:
I have updated code to match suggestion. Makes it simpler for sure.